backend/apps/onboarding/services/ml_predictor.py

```
@@ -0,0 +1,130 @@
+        probability = self.model.predict_proba(X_scaled)[0][1]
+        confidence = max(self.model.predict_proba(X_scaled)[0])
+
+        return float(probability), float(confidence)
+
+    def _extract_features(self, journey: OnboardingJourney) -> list:
+        """Extract features for prediction."""
+        events = JourneyEvent.objects.filter(journey=journey)
+
+        time_in_stage = 0
+        if journey.stage_started_at:
+            time_in_stage = (timezone.now() - journey.stage_started_at).total_seconds() / 3600
```

Ruff (0.15.21)

[error] Undefined name `timezone`

(F821)